### ISSUE TITLE
Multizone Support

### ISSUE THREAD

**Issue by McMichaeli** [imcfarla2003/domoticz-hive#2]
Is the current plug-in intended to support Hive Multizone heating installations where more than one thermostat and receiver are installed?

I have installed commit 23b4d3b on Domoticz v3.8153 on my Raspberry Pi.  It works just fine and locates my upstairs heating zone, which also has the hot water control on it.  However the downstairs zone is missing from the list of devices it adds.

I have run collect_json.py against my Hive account and can see both sets of details in the JSON that is returned.  Maybe this could be added as a feature request?

Thanks in advance,

Ian.

**Comment by imcfarla2003** [comment]
can you provide the output of collect_json.py?

**Comment by gnunan** [comment]
Hi Iain,

Here is the output that I got from your latest (as at 14/12/18) collect_json.py:

[collect.txt](https://github.com/imcfarla2003/domoticz-hive/files/2682491/collect.txt)

I have 3 heating zones - "Gnd Floor", "Mid Floor" and "Top Floor". The hot water is associated with "Gnd Floor" if that is relevant/helpful.

Many thanks,

Greg.

**Comment by imcfarla2003** [comment]
Can you try the latest commit please?

**Comment by gnunan** [comment]
Hi Iain,

Unfortunately still only the 1 zone (see attached jpeg of the devices):

![hive1](https://user-images.githubusercontent.com/45892975/50055379-ff5a7400-0145-11e9-9e9f-23259ed82296.jpg)

Greg.

**Comment by gnunan** [comment]
... oops, and here's the log:

[Hive1.log](https://github.com/imcfarla2003/domoticz-hive/files/2683677/Hive1.log)

Greg.

**Comment by imcfarla2003** [comment]
Can you try the latest commit please?

**Comment by gnunan** [comment]
Hi again Iain,

Here are the results showing the latest devices:

![hive2](https://user-images.githubusercontent.com/45892975/50123289-0ec9e200-0258-11e9-94e6-d16ee1d88e34.jpg)

and log:

[Hive2.log](https://github.com/imcfarla2003/domoticz-hive/files/2688269/Hive2.log)

Cheers,

Greg.

**Comment by imcfarla2003** [comment]
No errors - good

Have you set Domoticz to allow the creation of new devices?

Regards,

Iain

**Comment by imcfarla2003** [comment]
Can you also try python3 test.py <Hive username> <Hive password> from within the test folder and attach the output from that?

**Comment by gnunan** [comment]
Hi,

Yes, new devices are allowed (and Idxs 5 and 6 in the previous screenshot were created). I tried deleting the Hive hardware and re-added it and, lo and behold, all the devices appeared:

![hive3](https://user-images.githubusercontent.com/45892975/50148899-9c85eb80-02b1-11e9-824c-887dc08002b3.jpg)

with the following log:

[Hive3.log](https://github.com/imcfarla2003/domoticz-hive/files/2690102/Hive3.log)

but the Data fields look to be wrong this time (but were ok a bit later on).

I ran test.py as requested and got this output:

[test3.log](https://github.com/imcfarla2003/domoticz-hive/files/2690108/test3.log)

Note that I am running Domoticz version 4.9700 (the latest stable release) so I wonder if there is some issue with initially creating devices (I assume that there was no reason why the Idx 5 and 6 devices shouldn't have been created before) and/or displaying data for new devices?

So, it looks as though it is all working :-)

A cheeky request - would it be possible to put the name from the Hive data (if it's there) into the device Name in some way? It would make it a bit easier to initially identify - but of course it could be done manually too).

Many thanks,

Greg.
